src/core/symbology-ng/qgssymbolv2.cpp

```diff
@@ -903,27 +965,9 @@ void QgsSymbolV2::renderFeature( const QgsFeature& feature, QgsRenderContext& co
 
   if ( drawVertexMarker )
   {
-    const QgsCoordinateTransform* ct = context.coordinateTransform();
-    const QgsMapToPixel& mtp = context.mapToPixel();
-
-    QgsPointV2 vertexPoint;
-    QgsVertexId vertexId;
-    double x, y, z;
-    QPointF mapPoint;
-    while ( geom->geometry()->nextVertex( vertexId, vertexPoint ) )
+    Q_FOREACH ( const QPointF& marker, markers )
```

Comment:
@ahuarte47 Note that for this one, you actually want a copy rather then the reference. QPointF is small and it's quicker to copy than resolve a reference. So this should be `Q_FOREACH( QPointF marker, markers )` (this rule basically only applies to small inbuilt types like int, double, etc AND some very small Qt objects like QPoint, QPointF, QSize and QSizeF)